tests/phpunit/TestFramework/PhpUnit/Coverage/IndexXmlCoverageParserTest.php

```diff
@@ -65,14 +64,15 @@ final class IndexXmlCoverageParserTest extends TestCase
     private static $xml;
 
     /**
-     * @var LegacyXmlCoverageParserAdapter
+     * @var IndexXmlCoverageParser
      */
     private $parser;
 
     protected function setUp(): void
     {
-        $this->parser = new LegacyXmlCoverageParserAdapter(
-            new IndexXmlCoverageParser(XmlCoverageFixtures::FIXTURES_COVERAGE_DIR)
+        $this->parser = new IndexXmlCoverageParser(
+            XmlCoverageFixtures::FIXTURES_COVERAGE_DIR,
+            new XmlCoverageParser()
```

Comment:
Again, dependence of IndexXmlCoverageParser on XmlCoverageParser makes this test integration, where it could have been pure unit.

Reply:
We're already testing `XmlCoverageParser` in a dedicated test, so if to simplify `IndexXmlCoverageParser`,  this testing of `XmlCoverageParser` inside `IndexXmlCoverageParserTest` should be removed as redundant.

Reply:
I'm not changing anything there: I just got rid of the "LegacyAdapter" nothing more. I'm not saying there is further refactoring to do, e.g. test more those `CoverageReport` produced in `XmlCoverageParserTest` instead of `IndexXmlCoverageParserTest`, but this is not the goal of this PR

Reply:
I wish I could read them, and we'd avoid this conversation.